### ISSUE TITLE
Additional space in vertical layout

### ISSUE THREAD

**Issue by Fengson** [patchthecode/JTAppleCalendar#499]
I am using latest version of a library. I cannot find this issue in tutorials and sample codes.
When using vertical scrolling, some months take 6 rows and others 5. For the ones that take 6 everything looks fine. However, for the ones that take 5 rows, the additional row is being added and it's empty resulting in extra space.

This only happens if I want to hide cells that are not from this month. I use:
`cell.isHidden = cellState.dateBelongsTo != .thisMonth`

According to tutorial I just hide them:
https://www.youtube.com/watch?v=1Sta3aEo9E4

Results:
Sometimes it's rendered correctly:
<img width="323" alt="good" src="https://user-images.githubusercontent.com/5645565/28583315-0c23936a-7169-11e7-81a9-1c91635ff13c.png">

Other times not (additional row):
<img width="324" alt="bad" src="https://user-images.githubusercontent.com/5645565/28583340-1fe9fe98-7169-11e7-82a9-033cd06ee249.png">

Any ideas on how to fix this and remove additional empty row? Maybe I'm hiding cells incorrectly?

EDIT: Little debugging shows the problem. I replace the code:
`cell.isHidden = cellState.dateBelongsTo != .thisMonth`
with:
```
if cellState.dateBelongsTo == .thisMonth {
    cell.dateLabel.textColor = .blue
} else {
    cell.dateLabel.textColor = .red
}
```
and I get:
<img width="324" alt="good-debug" src="https://user-images.githubusercontent.com/5645565/28583881-0aa2d1e8-716b-11e7-9836-948072114847.png">
<img width="325" alt="bad-debug" src="https://user-images.githubusercontent.com/5645565/28583883-0efb9c48-716b-11e7-807d-6fa0bc0ba0a9.png">

Basically there is a whole row of days "not from this month" and then they are duplicated for next month, but hiding them results in an extra space. Any ideas how to fix it?

**Comment by patchthecode** [comment]
@Fengson so basically you do not want to hide then. You want them to not be generated at all.

in your `configureCalendar` function you have to return a `ConfigurationParameter` object to the library. Set the outDateGeneration to = `.endOrRow`.  
The layouts vid explain more on this [Here](https://www.youtube.com/watch?v=1Sta3aEo9E4)

**Comment by Fengson** [comment]
That was it, thanks!
